tls/s2n_cipher_suites.c

```diff
@@ -1056,7 +1056,7 @@ S2N_RESULT s2n_cipher_suites_cleanup(void)
     return S2N_RESULT_OK;
 }
 
-S2N_RESULT s2n_cipher_suite_from_iana(const uint8_t iana[static S2N_TLS_CIPHER_SUITE_LEN], struct s2n_cipher_suite **cipher_suite)
+S2N_RESULT s2n_cipher_suite_from_iana(const uint8_t *iana, struct s2n_cipher_suite **cipher_suite)
```

Comment:
This is unfortunate. I'm on the fence about removing this for a unit test. Changing the signature to just a pointer means the length is implied, which has other issues. With the array type we'd at least get a warning from supported compilers that the data behind the pointer is undersized.

Reply:
I am also not certain removing this array type is a good solution.

Reply:
I don't know. The reason I'm leaning away from removing the test is that we somehow managed to write and merge a unit test that passed a NULL to s2n_cipher_suite_from_iana, so apparently our CI isn't preventing that.

If this had been a compiler error instead of a seg fault, yes, I'd probably agree that the test should be removed. But the seg fault is concerning :\